core/orchestrator.go

```diff
@@ -254,6 +255,60 @@ func (orch *orchestrator) TicketParams(sender ethcommon.Address, priceInfo *net.
 	}, nil
 }
 
+func (orch *orchestrator) GetCapabilitiesPrices(sender ethcommon.Address) ([]*net.PriceInfo, error) {
+	ethAddr := sender.String()
+
+	//Orchestrators can have two prices for capability/model id
+	//if a price is set for a specific eth address, it will override the default price
+	defaultPrices := orch.node.GetCapsPrices("default")
+	gatewayPrices := orch.node.GetCapsPrices(ethAddr)
+
+	capPricesMap := make(map[string]*net.PriceInfo)
+
+	var capPrices []*net.PriceInfo
+	if defaultPrices != nil {
+		for cap, price := range *defaultPrices {
+			for modelID, priceInfo := range price.modelPrices {
+				if priceInfo == nil {
+					continue
+				}
+				priceInt64, err := common.PriceToInt64(priceInfo.Value())
+				if err != nil {
+					glog.Errorf("error converting %v price for capability %v to int64 err=%w", "default", CapabilityNameLookup[cap], err)
+					continue
+				}
+				capPriceName := strconv.Itoa(int(cap)) + "_" + modelID
+				capPricesMap[capPriceName] = &net.PriceInfo{PricePerUnit: priceInt64.Num().Int64(), PixelsPerUnit: 1, Capability: uint32(cap), Constraint: modelID}
```

Comment:
I see you're still using `1` as denominator. I understand from your other comment that it's because you're afraid of overflowing int64. However, are we ok losing the precision here?

Reply:
ah dang, that was a copy error on the refactor. Will fix.